hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/AbstractHoodieClient.java

```diff
@@ -124,9 +128,30 @@ public HoodieWriteConfig getConfig() {
     return config;
   }
 
+  private void checkIndexTypeCompatible(String writeIndexType, String persistIndexType) {
+    if (writeIndexType.equals(persistIndexType)) {
+      return;
+    } else {
+      if ((persistIndexType.equals(HoodieIndex.IndexType.GLOBAL_BLOOM.name())
```

Comment:
It appears that this is currently allowing the following cases 

a) if we persisted global_index, then allow transitions to BLOOM, SIMPLE. 


Additionally, I think 

- We should error out if INMEMORY is every used in production code :) 
- GLOBAL_SIMPLE should be compatible with GLOBAL_BLOOM (and converse)